carbonserver/carbonserver.go

```diff
@@ -472,7 +472,10 @@ func (listener *CarbonserverListener) updateFileList(dir string) {
 		return
 	}
 
-	freeSpace := stat.Bavail * uint64(stat.Bsize)
+	var freeSpace uint64
+	if stat.Bavail >= 0 {
```

Comment:
Was looking at this again, and think this may be better as `stat.Bavail > 0`. If the value is zero, `n * 0` is always zero. So, `var freeSpace uint64` takes care of that.